tools/sphere.c

```diff
@@ -37,40 +37,39 @@ extra_init() {
 
 // Apply Slooper and Sloopee shadows, respectively
 int do_sloop( string arg ) {
+    if ( !arg ) {
```

Comment:
Checking for a missing arg is a very common pattern. The check I actually like is: 
```
if (!stringp(arg) || !strlen(trim(arg))) {
  return notify_fail(msg);
}
```

Technically, you only _have_ to check !arg if you're assuming you're being executed by the driver as the consequence of an add_action(), so what you wrote is correct for that. But it's a public function, what if someone else out there called me and passed a float or something? Should it happen...no; but could it happen...yes. Maybe someone writes their own command controller or something instead of using the driver's -- it doesn't matter, we should defend against it.

My check ensures the arg is a string and there's something in it besides whitespace. Otherwise, you get the message for a missing argument string. This logic is something we might want to use again, for other commands. This means it's a good candidate for going into a shared library of functions. Just like how you inherit AnsiTellCode for your ansi_tell() functions, we could make our own library and put this logic into it to be used by your sphere. This is the beauty of object-oriented programming.

If this doesn't 100% make sense to you, we should talk about it, because it's important.